test/e2e/node_reuse_test.go

```diff
@@ -509,15 +511,37 @@ func untaintNodes(clientSet *kubernetes.Clientset, nodes *corev1.NodeList, taint
 		Logf("Untainting node %v ...", nodes.Items[i].Name)
 		newNode, changed := removeTaint(&nodes.Items[i], taint)
 		if changed {
-			node, err := clientSet.CoreV1().Nodes().Update(ctx, newNode, metav1.UpdateOptions{})
-			Expect(err).To(BeNil(), "Failed to update nodes")
-			Logf("Node %v untainted", node.Name)
+			err := patchNode(clientSet, nodes.Items[i].Name, &nodes.Items[i], newNode)
+			Expect(err).To(BeNil(), "Failed to patch node")
+			Logf("Node %v untainted", nodes.Items[i].Name)
 			count++
 		}
 	}
 	return
 }
 
+func patchNode(clientSet *kubernetes.Clientset, nodeName string, oldNode *corev1.Node, newNode *corev1.Node) error {
```

Comment:
I think it would be better to change this function to something more generic, e.g. `generatePatch(oldNode *corev1.Node, newNode *corev1.Node)`. And then call `Patch` instead of `Update` with the generated patch.
As it is now. `patchNode` does something very specific (and a bit surprising I think): it generates a patch for the taints and then applies it, but it ignores all other differences there could be between the Nodes.

Reply:
Well it could be used for patching labels or annotations also, not just taints. All I'm saying is that I would expect a function like `patchNode` to include everything, not just taints.

Reply:
It is considering only taints because we are trying to avoid conflict issues like `Operation cannot be fulfilled on nodes \"test1-wjvvm\": the object has been modified; please apply your changes to the latest version and try again` https://jenkins.nordix.org/view/Metal3%20Main/job/metal3_main_v1a5_e2e_test_ubuntu/22/console 
I would prefer to rename the function to something like `patchNodeTaints`

Reply:
Ok I'm fine with that :+1:

Reply:
done